core/src/calleeFunctions/index.ts

```diff
@@ -23,27 +23,101 @@ const allCalleeFunctions: Record<CalleeNames, CalleeFunctions> = {
+    return await allCalleeFunctions[marketData.callee].getCalleeData(network, collateral, marketId, profitAddress);
 };
 
-const _getMarketPrice = async function (
+export const getMarketData = async function (
     network: string,
     collateralSymbol: string,
     amount: BigNumber = new BigNumber('1')
-): Promise<BigNumber> {
+): Promise<Record<string, MarketData>> {
     const collateral = getCollateralConfigBySymbol(collateralSymbol);
-    const calleeFuctions = allCalleeFunctions[collateral.exchange.callee];
-    if (!calleeFuctions) {
+    const isCollateralSupported = Object.values(collateral.exchanges).some(
+        marketData => marketData.callee && allCalleeFunctions[marketData.callee]
+    );
+    if (!isCollateralSupported) {
         throw new Error(`Unsupported collateral symbol "${collateralSymbol}"`);
     }
-    return await calleeFuctions.getMarketPrice(network, collateral, amount);
+    let marketDataRecords = {};
+    for (const marketId in collateral.exchanges) {
+        const marketData = collateral.exchanges[marketId];
+        let marketUnitPrice: BigNumber;
+        try {
+            marketUnitPrice = await allCalleeFunctions[marketData.callee].getMarketPrice(
+                network,
+                collateral,
+                marketId,
+                amount
+            );
+        } catch {
+            marketUnitPrice = new BigNumber(NaN);
+        }
+        if (marketData.callee === 'UniswapV2LpTokenCalleeDai') {
+            marketDataRecords = {
+                ...marketDataRecords,
+                [marketId]: {
+                    marketUnitPrice,
+                    token0: marketData.token0,
+                    token1: marketData.token1,
+                },
+            };
+        } else {
+            marketDataRecords = {
+                ...marketDataRecords,
+                [marketId]: {
+                    marketUnitPrice,
+                    route: marketData.route,
+                },
+            };
+        }
+    }
+    return marketDataRecords;
+};
+
+export type BestMarketData = {
+    marketId: string;
+    marketUnitPrice: BigNumber;
+};
+
+export const getBestMarketData = async function (
+    marketDataRecords: Record<string, MarketData>
+): Promise<BestMarketData> {
+    const marketDataRecordsSorted = Object.entries(marketDataRecords);
+    marketDataRecordsSorted.sort((a, b) => {
+        // push NaNs to the end
+        if (a[1].marketUnitPrice.isNaN() && b[1].marketUnitPrice.isNaN()) {
+            return 1;
+        }
+        if (a[1].marketUnitPrice.isNaN()) {
+            return 1;
+        }
+        if (b[1].marketUnitPrice.isNaN()) {
+            return -1;
+        }
+        return a[1].marketUnitPrice.minus(b[1].marketUnitPrice).toNumber();
+    });
+    return {
+        marketId: marketDataRecordsSorted[0][0],
+        marketUnitPrice: marketDataRecordsSorted[0][1].marketUnitPrice,
+    };
```

Comment:
Nit: Maybe it might actually make sense to return just the `marketId` from this function (renaming it to `getBestMarketId`) or the whole market data. Otherwise the name is misleading